app/services/storage.js

```diff
@@ -29,13 +29,31 @@ export default Service.extend({
     return !!this.getItem('travis.auth.become');
   },
 
+  get billingStep() {
+    return +this.getItem('travis.billing_step');
+  },
+
+  get billingInfo() {
+    return this.getItem('travis.billing_info');
+  },
+
+  get billingPlan() {
+    return this.getItem('travis.selected_plan');
+  },
```

Comment:
This is neat! 😍The only thing I'd like to suggest here is to incapsulate the entire serialization/deserialization logic here, in the service, like this:
```js
get billingStep() {
  return +this.getItem('travis.billing_step');
},
set billingStep(value) {
  this.setItem('travis.billing_step', value);
},

get billingInfo() {
  try {
    return JSON.parse(this.getItem('travis.billing_info'));
  } catch(e) {
    return {};
  }
},
set billingInfo(value) {
  this.setItem(JSON.stringify('travis.billing_info'));
},
...
```
In this case we'll have all the serialization/deserialization logic in one place. And also the storage keys will also be in one place, so it'll be easy to change them if we'd ever need it.

Then in the external code you can just 
```js
const billingData = this.storage.billingInfo;
billingData.period = 'monthly';
this.storage.billingInfo = billingData;
```
And it will be serialized properly to the storage.

What do you think?

Reply:
Brilliant! 🎉 Thanks for the suggestion & hint. This is much more maintainable, and also means less Strings all over the place. :)